### ISSUE TITLE
flaky io_uring test: renameat

### ISSUE THREAD

**Issue by locriacyber** [ziglang/zig#10466]
### Zig Version

0.10.0-dev.40+303bad998

### Steps to Reproduce

Enter Zig repository and see below

### Expected Behavior
It should show what the error is.

### Actual Behavior

```
> zig test lib/std/os.zig --zig-lib-dir lib
Test [55/67] os.linux.io_uring.test "renameat"... thread 2944 panic: unhandled errno: E.BADF
/home/user/computing/libs/zig/lib/std/debug.zig:254:22: 0x298050 in std.debug.panicExtra (test)
    std.builtin.panic(msg, trace);
                     ^
/home/user/computing/libs/zig/lib/std/debug.zig:230:15: 0x28312a in std.debug.panic (test)
    panicExtra(null, format, args);
              ^
/home/user/computing/libs/zig/lib/std/os/linux/io_uring.zig:2475:40: 0x2330c5 in os.linux.io_uring.test "renameat" (test)
        else => |errno| std.debug.panic("unhandled errno: {}", .{errno}),
                                       ^
/home/user/computing/libs/zig/lib/std/special/test_runner.zig:80:28: 0x25fff3 in std.special.main (test)
        } else test_fn.func();
                           ^
/home/user/computing/libs/zig/lib/std/start.zig:543:22: 0x2589ec in std.start.callMain (test)
            root.main();
                     ^
/home/user/computing/libs/zig/lib/std/start.zig:495:12: 0x239e0e in std.start.callMainWithArgs (test)
    return @call(.{ .modifier = .always_inline }, callMain, .{});
           ^
/home/user/computing/libs/zig/lib/std/start.zig:409:17: 0x238ea6 in std.start.posixCallMainAndExit (test)
    std.os.exit(@call(.{ .modifier = .always_inline }, callMainWithArgs, .{ argc, argv, envp }));
                ^
/home/user/computing/libs/zig/lib/std/start.zig:322:5: 0x238cb2 in std.start._start (test)
    @call(.{ .modifier = .never_inline }, posixCallMainAndExit, .{});
    ^
error: the following test command crashed:
lib/std/zig-cache/o/9a9b5015cb232515c53314b45cdd126a/test /opt/zig/zig
```

**Comment by vrischmann** [comment]
Hi, what kernel version is this ? `renameat` is only available on kernel 5.11+, I thought [this check](https://github.com/ziglang/zig/blob/master/lib/std/os/linux/io_uring.zig#L2473-L2474) would be enough (the kernel version <5.11+ I tested on returned `EINVAL`).

Long term I think it would be better to use the probing feature of io_uring instead of adhoc checks.

**Comment by vrischmann** [comment]
I took a quick look, kernel 5.4 returns EBADF while 5.10 returns EINVAL.
